Update API overview page

gctucker opened the pull request:
Update the API overview page with more information and add an embedded Atlas graph with the number of nodes.

gctucker commented:
Deployed on staging: https://static.staging.kernelci.org/docs/api/

pawiecz commented on doc/_index.md:
```diff
@@ -1,13 +1,85 @@
+
+Each node object also has a state which can be used when orchestrating the
+pipeline.  For example, a node will be in the Running state while awaiting some
+results.  There's also a result value, to tell whether the job passed or
```
What is a "job" in this context? A generic task to be run, K8s job, LAVA job or something else?

gctucker replied:
See other comment below.  I'll resolve this one as it's just another occurrence of the same issue.

pawiecz commented on doc/_index.md:
```diff
@@ -1,13 +1,85 @@
+<div style="padding: 0 0 20px 0"><iframe style="background: #FFFFFF;border: none;border-radius: 2px;box-shadow: 0 2px 10px 0 rgba(70, 76, 79, .2);" width="640" height="480" src="https://charts.mongodb.com/charts-kernelci-api-staging-otxuy/embed/charts?id=64c9fa16-ba81-40e9-8cd7-cd56fb88fbf3&maxDataAge=3600&theme=light&autoRefresh=true"></iframe></div>
+
+> **Note:** The API doesn't manage storage, the only requirement is to provide
+> publicly-available HTTP(s) URLs for each artifact.
```
minor: typo: s/(s)/(S)/ ?

gctucker replied:
Yes I think I've seen it written as HTTP(s) before but as it's either HTTP or HTTPS I think a capital letter makes more sense.

pawiecz commented on doc/_index.md:
```diff
@@ -1,13 +1,85 @@
+
+### Publisher / Subscriber Interface
+
+Every time some data changes in the database, basically every time a node is
```
minor: typo: s/ is$// ?

pawiecz commented on doc/_index.md:
```diff
@@ -1,13 +1,85 @@
+
+Interacting with the API requires a token which is associated with a particular
+user.  Whenever a user sends some data such as a new node, it is owned by that
+user.  While all nodes are publicly readable, only the owner of the node can
```
I suppose admin has superuser rights which overrides this rule - should it be stated explicitly or is it irrelevant in this context?

gctucker replied:
I don't think that's relevant here as this is an overview - and in fact there appears to be a bug so even a user in the "admin" group can't update nodes apparently, see https://github.com/kernelci/kernelci-pipeline/issues/315

pawiecz commented on doc/_index.md:
```diff
@@ -1,13 +1,85 @@
+
+## Pipeline
+
+The Pipeline is all the client-side services which are running the actual jobs.
```
I might be missing some context (same as in L27) on what a "job" is.

I remember there was an attempt to make [test terms glossary](https://elinux.org/Test_Glossary). There's a "Run (noun)" term for what I believe a "job" in KCI API context might be - is it correct? Also, I'm not sure how widely this glossary is used.

gctucker replied:
It's true that it's not very well defined yet.  I'll rewrite this with some paraphrasing to avoid just using the word "job" as a wildcard.

Basically it's what you get when you run `kci job submit` etc. but it's very implicit for an overview documentation page.

gctucker replied:
I think a "run" is a close match but it's a bit unclear in my opinion.  Also "job" means something specific in Kubernetes so it's probably better to avoid using it here to avoid confusion.  And yes while this glossary is very nicely put together I don't think it's an authoritative standard across the whole ecosystem so we can't just assume that everyone will follow the same definitions.

gctucker replied:
Alright I've rephrased these parts and now there's no mention of "job" on this page.  I've replaced it with things like "pipeline steps", basically it's about what the pipeline effectively does.  Please let me know if this is OK.

pawiecz commented on doc/_index.md:
```diff
@@ -1,13 +1,85 @@
+any artifacts to some independent storage services and provide public URLs to
+access them.
+
+A standart set of pipeline services is run directly by KernelCI alongside the
```
minor: typo s/standart/standard/ ?

pawiecz replied:
```suggestion
A standard set of pipeline services is run directly by KernelCI alongside the
```

pawiecz commented:
I forgot I should have added code suggestions for small changes, let me update the comments.

gctucker commented:
Thanks for the review, I think I've addressed all the comments.  I'll now add another page about how to get an account and start using the public API instance deployed in AKS etc.  I think I'll merge this on Monday anyway to have the documentation ready for communicating about the beginning Early Access phase but it's always possible to add comments later on and I'll make follow-up PRs to fix anything else if needed.
